src/senaite/patient/content/patient.py

```diff
@@ -763,3 +767,17 @@ def setAddress(self, value):
         """
         mutator = self.mutator("address")
         return mutator(self, value)
+
+    def getFormattedAddress(self, atype=PHYSICAL_ADDRESS):
+        """Returns the formatted address
+        """
+        if atype not in POSSIBLE_ADDRESSES:
+            return None
+
+        address_format = patient_api.get_patient_address_format()
+
+        records = self.getAddress()
+        for record in records:
+            if atype != record.get("type"):
+                continue
+            return Template(address_format).safe_substitute(record)
```

Comment:
Note that if the address format is "$address, $zip $city, $country", and only the address has been entered, the returned output is `Somwhere in the middle of nowhere, ,". I suggest to handle special char separators like comma, and semicolon properly to generate a cleaner output. Something like:

```python
output = Template(address_format).safe_substitute(record)
output = filter(None, output.split(", "))
return ", ".join(output)
```

Reply:
Implemented in e6b51a7